### PR TITLE
Prompt user to enable Location if not enabled

### LINKED ISSUE #4987: GeoPoint does nothing if Location is disabled at a system level
#### Software and hardware versions 

Collect v2022.1 Beta 2

#### Problem description

Trying to to use the new GeoPoint with Location disabled in Android just causes the dialog to wait indefinitely (a location is never fixed and there is no error message).

#### Steps to reproduce the problem

1. Make sure Collect has location permissions granted
2. Disable Location in Android by going to "Settings" > "Location" and untoggling "Use Location"
3. Open a form with a GeoPoint (without appearance) question and navigate to it
4. Click on "Start GeoPoint"

#### Expected behavior

Collect should show an error message and ask the user to enable location/gps. We already do this for other geo widgets in the same scenario (`geopoint` with `placement-map` appearance for instance).

v2021.3.4 behaves weirdly (in my opinion) in this case: clicking "Save GeoPoint" results in the user being jumped to "Location" settings with a Toast (that's easy to missing) saying "Sorry, Location providers are disabled!". I think we should get closer to the behaviour we have in `placement-map` and show a dialog allowing the user to move to settings if they want.

### PR DESCRIPTION
**Pull request by seadowg**
Closes #4987 

I've made is Location services are checked after Location permissions and the user should only get to the GeoPoint dialog if the permissions are granted **and** Location is enabled now.

#### What has been done to verify that this works as intended?

Verified manually.

#### Why is this the best possible solution? Were any other approaches considered?

We discussed different places to put this check and decided that we should incorporate it in `PermissionsProvider` so we could have the case covered whenever location permissions are requested. I didn't want to add the risk of changing everything now (during regression testing) so I've introduced a new method that we can switch to later (I'll create an issue).

I also left out tests here as I think we need to rework how we test `PermissionsProvider` - it currently isn't easy to test permission requests without a spy. Let's cover that when we switch over to using this method everywhere.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Just changes the permissions flow for the default GeoPoint widget. It'd be good to try different combinations of permissions granted/denied and location services enabled/disabled. I think this problem caught us all by surprise so good for everyone to keep in mind that disabling "Location" in Android is different from denying location permissions.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/androidTest/java/org/odk/collect/android/support/pages/FormHierarchyPage.java [hunk 1]
```diff
@@ -21,6 +21,8 @@ public FormHierarchyPage(String formName) {
     @Override
     public FormHierarchyPage assertOnPage() {
         assertToolbarTitle(formName);
+        assertText(R.string.jump_to_beginning);
```
Comment (seadowg) [inline collect_app/src/androidTest/java/org/odk/collect/android/support/pages/FormHierarchyPage.java:24]: Before, the assertion was the same as the one in `FormEntryPage` which looks like it caused a timing issue in a Test Lab run I did.

**Review by grzesiek2010 — APPROVED** [review]

**Comment by kkrawczyk123** [comment]
**Tested with success!**
Verified on Androids: 5.1, 8.1 and 10

Verified cases:
- displayed dialog
- light/dark mode
- cancel button
- Enable GPS button
- minimize app
- enable GPS in device
- able to save geo point

One thing that I am not a fan of is that dialog disappears when rotating the screen but it's more nice to have than a bug.

**Comment by seadowg** [comment]
> One thing that I am not a fan of is that dialog disappears when rotating the screen but it's more nice to have than a bug.

It should pop up when they hit the button again though right? I think in general we've been trying to avoid worrying about dialogs staying on screen during rotation (unless they're blocking like progress dialogs) as it's pretty unlikely to happen in real life.

**Comment by srujner** [comment]
**Tested with success!**
Verified all the cases on Android 11

**Pull request merged**